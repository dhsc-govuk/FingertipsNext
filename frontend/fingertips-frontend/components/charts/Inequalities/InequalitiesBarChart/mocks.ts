import { InequalitiesBarChartData } from '@/components/charts/Inequalities/helpers/inequalitiesHelpers';

export const getTestData = (): InequalitiesBarChartData => ({
  areaCode: 'A1425',
  areaName: 'South FooBar',
  data: {
    period: 2008,
    inequalities: {
      Male: {
        value: 890.328253,
        count: 131,
        upper: 578.32766,
        lower: 441.69151,
        isAggregate: false,
      },
      Persons: {
        value: 135.149304,
        count: 222,
        upper: 578.32766,
        lower: 441.69151,
        isAggregate: true,
      },
      Female: {
        value: 890.328253,
        count: 131,
        upper: 578.32766,
        lower: 441.69151,
        isAggregate: false,
      },
    },
  },
});
