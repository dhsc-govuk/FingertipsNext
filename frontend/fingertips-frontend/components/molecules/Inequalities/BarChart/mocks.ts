import { InequalitiesBarChartData } from '@/components/organisms/Inequalities/inequalitiesHelpers';

export const getTestData = (): InequalitiesBarChartData => ({
  areaName: 'South FooBar',
  data: {
    period: 2008,
    inequalities: {
      Male: {
        value: 890.328253,
        count: 131,
        upper: 578.32766,
        lower: 441.69151,
      },
      Persons: {
        value: 135.149304,
        count: 222,
        upper: 578.32766,
        lower: 441.69151,
      },
      Female: {
        value: 890.328253,
        count: 131,
        upper: 578.32766,
        lower: 441.69151,
      },
    },
  },
});
