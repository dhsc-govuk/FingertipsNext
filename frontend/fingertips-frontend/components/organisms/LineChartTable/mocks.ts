export const MOCK_ENGLAND_DATA = {
  areaCode: 'E92000001',
  areaName: 'England',
  healthData: [
    {
      year: 2004,
      count: 200,
      value: 904.874,
      lowerCi: 0,
      upperCi: 0,
      ageBand: '0-4',
      sex: 'Female',
    },
    {
      year: 2008,
      count: 500,
      value: 965.9843,
      lowerCi: 0,
      upperCi: 0,
      ageBand: '10-14',
      sex: 'Female',
    },
  ],
};

export const MOCK_HEALTH_DATA = [
  {
    areaCode: 'A1425',
    areaName: 'Greater Manchester ICB - 00T',
    healthData: [
      {
        year: 2008,
        count: 222,
        value: 890.305692,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Persons',
      },
      {
        year: 2004,
        count: 267,
        value: 703.420759,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Persons',
      },
    ],
  },
  {
    ...MOCK_ENGLAND_DATA,
  },
  {
    areaCode: 'A1426',
    areaName: 'South FooBar',
    healthData: [
      {
        year: 2004,
        count: 222,
        value: 135.149304,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Persons',
      },
      {
        year: 2008,
        count: 131,
        value: 890.328253,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Persons',
      },
    ],
  },
];
