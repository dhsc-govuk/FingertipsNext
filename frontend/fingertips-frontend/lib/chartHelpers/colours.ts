import { GovukColours } from '../styleHelpers/colours';

export type ChartColours = GovukColours | UniqueChartColours;

export enum UniqueChartColours {
  ChartDarkBlue = '#12436D',
  OtherLightBlue = '#57AEF8',
}

// The ordering of this array determines the order in which the colours are
// used by Highcharts, so it should be preserved.
export const chartColours: ChartColours[] = [
  GovukColours.Yellow,
  GovukColours.Red,
  GovukColours.Blue,
  GovukColours.LightPink,
  GovukColours.Brown,
  UniqueChartColours.ChartDarkBlue,
  GovukColours.LightBlue,
  GovukColours.OtherLightPurple,
  GovukColours.Orange,
  GovukColours.LightPurple,
  GovukColours.DarkPink,
  GovukColours.Green,
  GovukColours.Pink,
  GovukColours.Purple,
];
