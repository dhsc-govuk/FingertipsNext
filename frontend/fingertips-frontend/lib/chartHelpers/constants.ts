export const areaCodeForEngland = 'E92000001';
export const englandAreaString = 'England';
export const indicatorIdForPopulation = 337;
