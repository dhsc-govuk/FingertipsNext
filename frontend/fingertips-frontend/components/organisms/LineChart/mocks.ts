import {
  HealthDataForArea,
  HealthDataPointTrendEnum,
} from '@/generated-sources/ft-api-client';
import { allAgesAge, noDeprivation, personsSex } from '@/lib/mocks';

export const mockIndicatorData: HealthDataForArea[] = [
  {
    areaCode: 'A1425',
    areaName: 'North FooBar',
    healthData: [
      {
        count: 389,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        value: 278.29134,
        year: 2006,
        periodLabel: '2006',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
      {
        count: 267,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        value: 703.420759,
        year: 2004,
        periodLabel: '2004',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
    ],
  },
  {
    areaCode: 'A1426',
    areaName: 'South FooBar',
    healthData: [
      {
        count: 654,
        lowerCi: 750.69151,
        upperCi: 800.32766,
        value: 786.27434,
        year: 2010,
        periodLabel: '2010',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
      {
        count: 567,
        lowerCi: 440.69151,
        upperCi: 420.32766,
        value: 435.420759,
        year: 2007,
        periodLabel: '2007',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
    ],
  },
  {
    areaCode: 'A1427',
    areaName: 'East FooBar',
    healthData: [
      {
        count: 953,
        lowerCi: 460.69151,
        upperCi: 500.32766,
        value: 478.27434,
        year: 2020,
        periodLabel: '2020',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
      {
        count: 567,
        lowerCi: 220.69151,
        upperCi: 250.32766,
        value: 234.420759,
        year: 2012,
        periodLabel: '2012',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
    ],
  },
];

export const mockBenchmarkData: HealthDataForArea = {
  areaCode: 'E92000001',
  areaName: 'England',
  healthData: [
    {
      count: 389,
      lowerCi: 441.69151,
      upperCi: 578.32766,
      value: 278.29134,
      year: 2006,
      periodLabel: '2006',
      sex: personsSex,
      ageBand: allAgesAge,
      trend: HealthDataPointTrendEnum.NotYetCalculated,
      deprivation: noDeprivation,
      isAggregate: true,
    },
    {
      count: 267,
      lowerCi: 441.69151,
      upperCi: 578.32766,
      value: 703.420759,
      year: 2004,
      periodLabel: '2004',
      sex: personsSex,
      ageBand: allAgesAge,
      trend: HealthDataPointTrendEnum.NotYetCalculated,
      deprivation: noDeprivation,
      isAggregate: true,
    },
  ],
};

export const mockParentData: HealthDataForArea = {
  areaCode: 'P001',
  areaName: 'Parent',
  healthData: [
    {
      count: 100,
      lowerCi: 200,
      upperCi: 400,
      value: 300,
      year: 2006,
      periodLabel: '2006',
      sex: personsSex,
      ageBand: allAgesAge,
      trend: HealthDataPointTrendEnum.NotYetCalculated,
      deprivation: noDeprivation,
      isAggregate: true,
    },
    {
      count: 101,
      lowerCi: 201,
      upperCi: 401,
      value: 301,
      year: 2004,
      periodLabel: '2004',
      sex: personsSex,
      ageBand: allAgesAge,
      trend: HealthDataPointTrendEnum.NotYetCalculated,
      deprivation: noDeprivation,
      isAggregate: true,
    },
  ],
};
