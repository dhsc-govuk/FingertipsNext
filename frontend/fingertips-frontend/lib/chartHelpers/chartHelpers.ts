import { HealthDataForArea } from '@/generated-sources/ft-api-client';
import { areaCodeForEngland } from './constants';

export function sortHealthDataByDate(
  data: HealthDataForArea[]
): HealthDataForArea[] {
  return data.map((item) => ({
    ...item,
    healthData: item.healthData.toSorted((a, b) => a.year - b.year),
  }));
}

export function sortHealthDataByYearDescending(
  data: HealthDataForArea[] = []
): HealthDataForArea[] {
  return data.map((item) => ({
    ...item,
    healthData: item.healthData.toSorted((a, b) => b.year - a.year),
  }));
}

export function seriesDataForIndicatorIndexAndArea(
  data: HealthDataForArea[][],
  indicatorIndex: number,
  seriesAreaCode: string
) {
  return data[indicatorIndex].find(
    (areaData) => areaData.areaCode === seriesAreaCode
  );
}

export function seriesDataWithoutEnglandOrParent(
  data: HealthDataForArea[],
  parentAreaCode?: string
) {
  return data.filter(
    (item) =>
      item.areaCode !== areaCodeForEngland && item.areaCode !== parentAreaCode
  );
}
