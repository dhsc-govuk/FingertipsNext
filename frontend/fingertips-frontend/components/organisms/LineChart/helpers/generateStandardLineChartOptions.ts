import {
  HealthDataForArea,
  BenchmarkComparisonMethod,
} from '@/generated-sources/ft-api-client';
import {
  sortHealthDataForAreasByDate,
  getFirstYearForAreas,
  getLatestYearForAreas,
  sortHealthDataForAreaByDate,
  AXIS_LABEL_FONT_SIZE,
} from '@/lib/chartHelpers/chartHelpers';
import { ChartColours, chartColours } from '@/lib/chartHelpers/colours';
import { areaCodeForEngland } from '@/lib/chartHelpers/constants';
import { SymbolKeyValue } from 'highcharts';
import { generateYAxis } from './generateYAxis';
import { generateXAxis } from './generateXAxis';
import { generateTooltip } from './generateTooltip';
import { generateAccessibility } from './generateAccessibility';
import { generateSeriesData } from './generateSeriesData';

export const lineChartDefaultOptions: Highcharts.Options = {
  credits: {
    enabled: false,
  },
  chart: {
    type: 'line',
    spacingBottom: 50,
    spacingTop: 20,
    animation: false,
    height: 500,
  },
  title: {
    style: {
      display: 'none',
    },
  },
  yAxis: {
    minorTickInterval: 'auto',
    minorTicksPerMajor: 2,
    labels: { style: { fontSize: AXIS_LABEL_FONT_SIZE } },
  },
  xAxis: {
    tickLength: 0,
    allowDecimals: false,
    labels: { style: { fontSize: AXIS_LABEL_FONT_SIZE } },
  },
  legend: {
    verticalAlign: 'top',
    align: 'left',
    itemStyle: {
      fontSize: '16px',
    },
    margin: 30,
  },
  accessibility: {
    enabled: false,
  },
  tooltip: {
    formatter: function (this: Highcharts.Point): string {
      return this.value?.toString() ?? '';
    },
  },
  plotOptions: {
    series: {
      animation: false,
    },
  },
};

export const chartSymbols: SymbolKeyValue[] = [
  'square',
  'triangle',
  'triangle-down',
  'circle',
  'diamond',
];

const determineBenchmarkData = (
  benchmarkToUse: string,
  englandData?: HealthDataForArea,
  groupIndicatorData?: HealthDataForArea
): HealthDataForArea | undefined => {
  if (benchmarkToUse === areaCodeForEngland && englandData) {
    return sortHealthDataForAreaByDate(englandData);
  } else if (benchmarkToUse !== areaCodeForEngland && groupIndicatorData) {
    return sortHealthDataForAreaByDate(groupIndicatorData);
  }
  return undefined;
};

const determineGroupData = (
  benchmarkToUse: string,
  englandData?: HealthDataForArea,
  groupIndicatorData?: HealthDataForArea
): HealthDataForArea | undefined => {
  if (benchmarkToUse === areaCodeForEngland && groupIndicatorData) {
    return sortHealthDataForAreaByDate(groupIndicatorData);
  } else if (benchmarkToUse !== areaCodeForEngland && englandData) {
    return sortHealthDataForAreaByDate(englandData);
  }
  return undefined;
};

export function generateStandardLineChartOptions(
  healthIndicatorData: HealthDataForArea[],
  lineChartCI: boolean,
  benchmarkToUse: string,
  optionalParams?: {
    englandData?: HealthDataForArea;
    groupIndicatorData?: HealthDataForArea;
    yAxisTitle?: string;
    xAxisTitle?: string;
    measurementUnit?: string;
    accessibilityLabel?: string;
    colours?: ChartColours[];
    symbols?: SymbolKeyValue[];
    xAxisLabelFormatter?: Highcharts.AxisLabelsFormatterCallbackFunction;
    benchmarkComparisonMethod?: BenchmarkComparisonMethod;
  }
): Highcharts.Options {
  const sortedHealthIndicatorData =
    sortHealthDataForAreasByDate(healthIndicatorData);
  const firstYear = getFirstYearForAreas(sortedHealthIndicatorData);
  const lastYear = getLatestYearForAreas(sortedHealthIndicatorData);

  const sortedBenchMarkData = determineBenchmarkData(
    benchmarkToUse,
    optionalParams?.englandData,
    optionalParams?.groupIndicatorData
  );

  const filteredSortedBenchMarkData =
    sortedBenchMarkData &&
    sortedHealthIndicatorData.length &&
    firstYear &&
    lastYear
      ? {
          ...sortedBenchMarkData,
          healthData:
            sortedBenchMarkData?.healthData.filter(
              (data) => data.year >= firstYear && data.year <= lastYear
            ) ?? [],
        }
      : sortedBenchMarkData;

  const sortedGroupData = determineGroupData(
    benchmarkToUse,
    optionalParams?.englandData,
    optionalParams?.groupIndicatorData
  );

  const filteredSortedGroupData =
    sortedGroupData && sortedHealthIndicatorData.length && firstYear && lastYear
      ? {
          ...sortedGroupData,
          healthData:
            sortedGroupData?.healthData.filter(
              (data) => data.year >= firstYear && data.year <= lastYear
            ) ?? [],
        }
      : sortedGroupData;

  return {
    ...lineChartDefaultOptions,
    yAxis: generateYAxis(optionalParams?.yAxisTitle),
    xAxis: generateXAxis(
      optionalParams?.xAxisTitle,
      optionalParams?.xAxisLabelFormatter
    ),
    series: generateSeriesData(
      sortedHealthIndicatorData,
      optionalParams?.symbols ?? chartSymbols,
      optionalParams?.colours ?? chartColours,
      filteredSortedBenchMarkData,
      filteredSortedGroupData,
      lineChartCI
    ),
    tooltip: generateTooltip(
      sortedHealthIndicatorData,
      sortedBenchMarkData,
      sortedGroupData,
      optionalParams?.benchmarkComparisonMethod,
      optionalParams?.measurementUnit
    ),
    accessibility: generateAccessibility(optionalParams?.accessibilityLabel),
  };
}
