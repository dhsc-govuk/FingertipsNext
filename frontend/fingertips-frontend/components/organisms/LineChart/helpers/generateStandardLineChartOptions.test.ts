/* eslint-disable @typescript-eslint/no-explicit-any */
import { HealthDataForArea } from '@/generated-sources/ft-api-client';
import { areaCodeForEngland } from '@/lib/chartHelpers/constants';
import { allAgesAge, personsSex, noDeprivation } from '@/lib/mocks';
import { generateStandardLineChartOptions } from './generateStandardLineChartOptions';
import { mockIndicatorData, mockBenchmarkData, mockParentData } from '../mocks';
import { GovukColours } from '@/lib/styleHelpers/colours';
import { SymbolKeyValue } from 'highcharts';

const symbols: SymbolKeyValue[] = ['arc', 'circle', 'diamond'];

const chartColours: GovukColours[] = [
  GovukColours.Orange,
  GovukColours.LightPurple,
  GovukColours.DarkPink,
];

describe('generateStandardLineChartOptions', () => {
  it('should generate standard line chart options', () => {
    const generatedOptions = generateStandardLineChartOptions(
      [mockIndicatorData[0]],
      false,
      areaCodeForEngland,
      {
        englandData: undefined,
        groupIndicatorData: undefined,
        yAxisTitle: 'yAxis',
        xAxisTitle: 'xAxis',
        measurementUnit: '%',
        accessibilityLabel: 'accessibility',
        colours: chartColours,
        symbols,
      }
    );

    expect((generatedOptions.yAxis as any)!.title.text).toBe('yAxis');
    expect((generatedOptions.xAxis as any)!.title.text).toBe('xAxis');
    expect(generatedOptions.accessibility!.description).toBe('accessibility');
    expect(generatedOptions).toMatchSnapshot();
  });

  it('should generate standard line chart options with benchmark and group data', () => {
    const generatedOptions = generateStandardLineChartOptions(
      [mockIndicatorData[0]],
      false,
      areaCodeForEngland,
      {
        englandData: mockBenchmarkData,
        groupIndicatorData: mockParentData,
        yAxisTitle: 'yAxis',
        xAxisTitle: 'xAxis',
        measurementUnit: '%',
        accessibilityLabel: 'accessibility',
        colours: chartColours,
        symbols,
      }
    );
    expect(generatedOptions).toMatchSnapshot();
  });

  it('should not include benchmark or group years before or after the areas have data', () => {
    const mockBenchmarkAreaWithEarlyYear: HealthDataForArea = {
      ...mockBenchmarkData,
      healthData: [
        ...mockBenchmarkData.healthData,
        {
          year: 1999,
          ageBand: allAgesAge,
          sex: personsSex,
          trend: 'Not yet calculated',
          deprivation: noDeprivation,
        },
      ],
    };
    const mockGroupAreaWithLateYear: HealthDataForArea = {
      ...mockParentData,
      healthData: [
        ...mockParentData.healthData,
        {
          year: 2036,
          ageBand: allAgesAge,
          sex: personsSex,
          trend: 'Not yet calculated',
          deprivation: noDeprivation,
        },
      ],
    };

    const generatedOptions = generateStandardLineChartOptions(
      [mockIndicatorData[0]],
      false,
      areaCodeForEngland,
      {
        englandData: mockBenchmarkAreaWithEarlyYear,
        groupIndicatorData: mockGroupAreaWithLateYear,
        yAxisTitle: 'yAxis',
        xAxisTitle: 'xAxis',
        measurementUnit: '%',
        accessibilityLabel: 'accessibility',
        colours: chartColours,
        symbols,
      }
    );
    expect((generatedOptions.series?.[0] as any).data).toHaveLength(2);
    expect((generatedOptions.series?.[1] as any).data).toHaveLength(2);
    expect(generatedOptions).toMatchSnapshot();
  });
});
