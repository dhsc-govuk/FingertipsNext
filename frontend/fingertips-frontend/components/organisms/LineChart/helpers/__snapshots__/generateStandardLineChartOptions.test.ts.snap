// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateStandardLineChartOptions should generate standard line chart options 1`] = `
{
  "accessibility": {
    "description": "accessibility",
    "enabled": false,
  },
  "chart": {
    "animation": false,
    "height": 500,
    "spacingBottom": 50,
    "spacingTop": 20,
    "type": "line",
  },
  "credits": {
    "enabled": false,
  },
  "legend": {
    "align": "left",
    "itemStyle": {
      "fontSize": "16px",
    },
    "margin": 30,
    "verticalAlign": "top",
  },
  "plotOptions": {
    "series": {
      "animation": false,
    },
  },
  "series": [
    {
      "color": "#F46A25",
      "custom": {
        "areaCode": "A1425",
      },
      "data": [
        [
          2004,
          703.420759,
        ],
        [
          2006,
          278.29134,
        ],
      ],
      "marker": {
        "symbol": "arc",
      },
      "name": "North FooBar",
      "type": "line",
    },
  ],
  "title": {
    "style": {
      "display": "none",
    },
  },
  "tooltip": {
    "formatter": [Function],
    "headerFormat": "",
    "useHTML": true,
  },
  "xAxis": {
    "allowDecimals": false,
    "labels": {
      "formatter": undefined,
      "style": {
        "fontSize": 16,
      },
    },
    "tickLength": 0,
    "title": {
      "margin": 20,
      "style": {
        "fontSize": 19,
      },
      "text": "xAxis",
    },
  },
  "yAxis": {
    "labels": {
      "formatter": [Function],
      "style": {
        "fontSize": 16,
      },
    },
    "minorTickInterval": "auto",
    "minorTicksPerMajor": 2,
    "title": {
      "margin": 20,
      "style": {
        "fontSize": 19,
      },
      "text": "yAxis",
    },
  },
}
`;

exports[`generateStandardLineChartOptions should generate standard line chart options with benchmark and group data 1`] = `
{
  "accessibility": {
    "description": "accessibility",
    "enabled": false,
  },
  "chart": {
    "animation": false,
    "height": 500,
    "spacingBottom": 50,
    "spacingTop": 20,
    "type": "line",
  },
  "credits": {
    "enabled": false,
  },
  "legend": {
    "align": "left",
    "itemStyle": {
      "fontSize": "16px",
    },
    "margin": 30,
    "verticalAlign": "top",
  },
  "plotOptions": {
    "series": {
      "animation": false,
    },
  },
  "series": [
    {
      "color": "#505A5F",
      "custom": {
        "areaCode": "E92000001",
      },
      "data": [
        [
          2004,
          703.420759,
        ],
        [
          2006,
          278.29134,
        ],
      ],
      "marker": {
        "symbol": "circle",
      },
      "name": "Benchmark: England",
      "type": "line",
    },
    {
      "color": "#28A197",
      "custom": {
        "areaCode": "P001",
      },
      "data": [
        [
          2004,
          301,
        ],
        [
          2006,
          300,
        ],
      ],
      "marker": {
        "symbol": "diamond",
      },
      "name": "Group: Parent",
      "type": "line",
    },
    {
      "color": "#F46A25",
      "custom": {
        "areaCode": "A1425",
      },
      "data": [
        [
          2004,
          703.420759,
        ],
        [
          2006,
          278.29134,
        ],
      ],
      "marker": {
        "symbol": "arc",
      },
      "name": "North FooBar",
      "type": "line",
    },
  ],
  "title": {
    "style": {
      "display": "none",
    },
  },
  "tooltip": {
    "formatter": [Function],
    "headerFormat": "",
    "useHTML": true,
  },
  "xAxis": {
    "allowDecimals": false,
    "labels": {
      "formatter": undefined,
      "style": {
        "fontSize": 16,
      },
    },
    "tickLength": 0,
    "title": {
      "margin": 20,
      "style": {
        "fontSize": 19,
      },
      "text": "xAxis",
    },
  },
  "yAxis": {
    "labels": {
      "formatter": [Function],
      "style": {
        "fontSize": 16,
      },
    },
    "minorTickInterval": "auto",
    "minorTicksPerMajor": 2,
    "title": {
      "margin": 20,
      "style": {
        "fontSize": 19,
      },
      "text": "yAxis",
    },
  },
}
`;

exports[`generateStandardLineChartOptions should not include benchmark or group years before or after the areas have data 1`] = `
{
  "accessibility": {
    "description": "accessibility",
    "enabled": false,
  },
  "chart": {
    "animation": false,
    "height": 500,
    "spacingBottom": 50,
    "spacingTop": 20,
    "type": "line",
  },
  "credits": {
    "enabled": false,
  },
  "legend": {
    "align": "left",
    "itemStyle": {
      "fontSize": "16px",
    },
    "margin": 30,
    "verticalAlign": "top",
  },
  "plotOptions": {
    "series": {
      "animation": false,
    },
  },
  "series": [
    {
      "color": "#505A5F",
      "custom": {
        "areaCode": "E92000001",
      },
      "data": [
        [
          2004,
          703.420759,
        ],
        [
          2006,
          278.29134,
        ],
      ],
      "marker": {
        "symbol": "circle",
      },
      "name": "Benchmark: England",
      "type": "line",
    },
    {
      "color": "#28A197",
      "custom": {
        "areaCode": "P001",
      },
      "data": [
        [
          2004,
          301,
        ],
        [
          2006,
          300,
        ],
      ],
      "marker": {
        "symbol": "diamond",
      },
      "name": "Group: Parent",
      "type": "line",
    },
    {
      "color": "#F46A25",
      "custom": {
        "areaCode": "A1425",
      },
      "data": [
        [
          2004,
          703.420759,
        ],
        [
          2006,
          278.29134,
        ],
      ],
      "marker": {
        "symbol": "arc",
      },
      "name": "North FooBar",
      "type": "line",
    },
  ],
  "title": {
    "style": {
      "display": "none",
    },
  },
  "tooltip": {
    "formatter": [Function],
    "headerFormat": "",
    "useHTML": true,
  },
  "xAxis": {
    "allowDecimals": false,
    "labels": {
      "formatter": undefined,
      "style": {
        "fontSize": 16,
      },
    },
    "tickLength": 0,
    "title": {
      "margin": 20,
      "style": {
        "fontSize": 19,
      },
      "text": "xAxis",
    },
  },
  "yAxis": {
    "labels": {
      "formatter": [Function],
      "style": {
        "fontSize": 16,
      },
    },
    "minorTickInterval": "auto",
    "minorTicksPerMajor": 2,
    "title": {
      "margin": 20,
      "style": {
        "fontSize": 19,
      },
      "text": "yAxis",
    },
  },
}
`;
